### ISSUE TITLE
"kaspawallet dump-encrypted-data" not working

### ISSUE THREAD

**Issue by thesheepcat** [kaspanet/kaspad#1851]
I have two servers (A and B) running Kaspa.

I need to migrate my wallet from A to B.

Server A:
- I installed the node
- I created a new wallet
- I used Kaspad v0.11.0 (binaries) and mined since Mainnet Day 1
- Then i updated the node to Kaspad v0.11.2 (binaries)
- Everything works correctly but i need to move the node on another server
- Therefore, i backuped KEYS.JSON and i dumped the seed by using ./kaspawallet dump-unencrypted-data
- The dump of the seed works correctly
- Before sutting down the server, i need to be sure that everything works on Server B

Server B:
- I installed Kaspad v0.11.2 (binaries)
- I created a new wallet
- I replaced KEYS.JSON (before bootstrapping the node) with KEYS.JSON of Server 1
- I bootstrapped the node without --utxoindex prefix
- Once synced, i stop the node and i activated again, adding --utxoindex prefix
- kapsawallet balance correctly works and gives me the right balance data
- Before shutting down Server A i wanted to check complete consistency and i tried to dump the seed on Server B
- kaspawallet dump-unencrypted-data ask for confirmation and ask for password
- I input the correct password but, everytime, it gives me the following error: "chacha20poly1305: message authentication failed"

I need to be sure the dump also works on Server B, before shutting down Server A.

![image](https://user-images.githubusercontent.com/68336151/141692695-ea4cdc88-1b3e-4474-8832-3c03a6856502.png)

**Comment by thesheepcat** [comment]
The situation is now even more tricky: 
- As both servers/nodes are still active, i'm mining on both of them;
- Both nodes are syncronized, both nodes are correctly mining on the same address, both nodes has the same wallet (keys.json).

The balance between the two node is different by 30% .

I already tried to restart the node which seems to be not showing the complete balance, without the --utxoindex flag; then, i stopped the node once synced; then, i re-run the node with --utxoindex parameter.

Still, the balances are different.

To summarize:
- Two nodes
- Same wallet
- Different balance

**Comment by dimdumon** [comment]
> 

Can you double-check please if ```dump-unencrypted-data``` still gives you exact same error ? Cause I tried it, and it worked for me correctly.

**Comment by someone235** [comment]
This is probably a side effect of the bug that's fixed in the dev branch https://github.com/kaspanet/kaspad/pull/1859

**Comment by thesheepcat** [comment]
> > 
> 
> Can you double-check please if `dump-unencrypted-data` still gives you exact same error ? Cause I tried it, and it worked for me correctly. My version is 0.11.6

Did you try to move your keys.json file (your wallet file) on a PC / VPS with a different number of cores (like from quad core PC to 6 core PC)?

**Comment by dimdumon** [comment]
No, I didn't - as far as I know, this was known issue and was fixed. Do you want me to try ? Does it related to current issue ?

________________________________
